### ISSUE TITLE
consider including WebRTC

### ISSUE THREAD

**Issue by louaybassbouss** [w3c-cg/webmediaapi#295]
WebRTC enables real-time communication (audio, video, data) on the Web and is already adopted in nearly any video conferencing  product. It became last year [W3C Recommendation](https://www.w3.org/TR/webrtc/) and is supported in all modern desktop and mobile browsers ([Can I use WebRTC](https://caniuse.com/?search=webrtc)). While the original focus of WebRTC is on video conferencing, it is increasingly used for ultra low latency (sub-second) live streaming e.g. for enabling interactivity. 

DASH-IF for example started an [activity](https://dashif.org/webRTC/) to explore synergies between DASH-based and WebRTC-based streaming. 

Should we consider WebRTC in WMAS2022?

**Comment by JohnRiv** [comment]
Note from this week's TWG meeting, if we include WebRTC, we should consider what that means and if a subset is more appropriate (given that most TVs do not have a camera nor a microphone attached to them).

**Comment by louaybassbouss** [comment]
> Note from this week's TWG meeting, if we include WebRTC, we should consider what that means and if a subset is more appropriate (given that most TVs do not have a camera nor a microphone attached to them).

Thanks @JohnRiv,  when I added this issue I was thinking about using WebRTC for sub-second latency media streaming  use case and not for real-time communication/video conferencing use case, which may be not suitable on TVs as you mentioned. For streaming, only the WebRTC API is needed. For real-time communication, you need also the [Media Capture and Streams API](https://www.w3.org/TR/mediacapture-streams/) to access camera and microphone. The [Screen Capture API](https://www.w3.org/TR/screen-capture/ ) can be used as well for to capture screen and share via WebRTC.  

To summarise, this issue in only about considering the WebRTC API without the Capturing APIs.

**Comment by JohnRiv** [comment]
March 2nd HATF Update - If we include WebRTC, we should mention if Media Capture is included or not (explicitly or if it is a dependency)

**Comment by acbegen** [comment]
FWIW, I agree, the webrtc relevance to the TVs is limited to streaming interactive content ad/or at low latency. Conferencing capabilities can be easily left out.

**Comment by acbegen** [comment]
Another comment is that the fresh-out-of-the-oven report from DASH-IF lists a bunch of open issues, things to be tackled and asks some tough questions in terms of where each of the open issues needs to be addressed. It will be good to get some answers from CE/TV vendors regarding the practicality of the features needed here. Based on that WMAS2022 might have a better answer as to whether include webrtc or not.

**Comment by louaybassbouss** [comment]
Additional Use Case I forgot to mention it when I created this issue is "Cloud Gaming". WebRTC plays a central role to enable cloud gaming on browser platforms and even on devices with limited processing capabilities. Cloud/Edge Rendering of complex applications like Metaverse Experiences and streaming over WebRTC is relevant as well.

**Comment by JohnRiv** [comment]
Proposed text to work form as a starting point if we were to include WebRTC in **3.5 Media specifications** of WMAS2022:

- WebRTC [[WEBRTC](https://www.w3.org/TR/webrtc/)]
  - Note: Since consumer products may not have camera and/or microphone inputs to send video and audio, WebRTC's functionality can be limited to not allow an [`RTCRtpTransceiverDirection`](https://www.w3.org/TR/webrtc/#dom-rtcrtptransceiverdirection) of "`sendrecv`" or "`sendonly`" on those devices. This also implies the [`RTCRtpSender` interface](https://www.w3.org/TR/webrtc/#rtcrtpsender-interface) is not required to be implemented, and the Media Capture and Streams [[GETUSERMEDIA](https://www.w3.org/TR/mediacapture-streams/)] requirements are limited to [`MediaStream`](https://www.w3.org/TR/mediacapture-streams/#dom-mediastream), [`MediaStreamTrack`](https://www.w3.org/TR/mediacapture-streams/#dom-mediastreamtrack), and [`MediaStreamConstraints`](https://www.w3.org/TR/mediacapture-streams/#dom-mediastreamconstraints) as defined in Media Capture and Streams and extended by WebRTC. This enables WebRTC to be leveraged for use cases such as security camera feeds, ultra low latency live streaming, and cloud gaming.

**Comment by JohnRiv** [comment]
Closed via #303
